qutebrowser/browser/webengine/notification.py

```diff
@@ -0,0 +1,134 @@
+        # to prevent it from getting GC'd. Otherwise, random methods start
+        # getting called with the notification as `self`, or segfaults happen,
+        # or other badness.
+        def _present_and_reset(qt_notification: QWebEngineNotification) -> None:
+            profile.setNotificationPresenter(_present_and_reset)
+            self._present(qt_notification)
+
+        profile.setNotificationPresenter(_present_and_reset)
+
+    def _present(self, qt_notification: QWebEngineNotification) -> None:
+        """Shows a notification over DBus.
+
+        This should *not* be directly passed to setNotificationPresenter
+        because of a bug in the PyQtWebEngine bindings.
+        """
+        # notification id 0 means 'assign us the ID'. We can't just pass 0
+        # because it won't get sent as the right type.
+        notification_id = QVariant(0)
+        notification_id.convert(QVariant.UInt)
+
+        actions_list = QDBusArgument([], QMetaType.QStringList)
+
+        qt_notification.show()
+        hints: Dict[str, Any] = {
+            # Include the origin in case the user wants to do different things
+            # with different origin's notifications.
+            "x-qutebrowser-origin": qt_notification.origin().toDisplayString()
+        }
+        if not qt_notification.icon().isNull():
+            hints["image-data"] = self._convert_image(qt_notification.icon())
+
+        reply = self.interface.call(
+            QDBus.BlockWithGui,
+            "Notify",
+            "qutebrowser",  # application name
+            notification_id,
+            "qutebrowser",  # icon
+            qt_notification.title(),
+            qt_notification.message(),
+            actions_list,
+            hints,
+            -1,  # timeout; -1 means 'use default'
+        )
+
+        if not (len(reply.arguments()) == 1):
+            raise DBusException(
+                "Got an unexpected number of reply arguments {}".format(
+                    len(reply.arguments())
+                )
+            )
+
+        notification_id = reply.arguments()[0]
+        log.sessions.debug("Sent out notification {}".format(notification_id))
+
+    def _convert_image(self, qimage: QImage) -> QDBusArgument:
+        """Converts a QImage to the structure DBus expects."""
+        # This is apparently what GTK-based notification daemons expect; tested it with dunst.
+        # Otherwise you get weird color schemes.
+        qimage.convertTo(QImage.Format_RGBA8888)
+        image_data = QDBusArgument()
+        image_data.beginStructure()
+        image_data.add(qimage.width())
+        image_data.add(qimage.height())
+        image_data.add(qimage.bytesPerLine())
+        image_data.add(qimage.hasAlphaChannel())
+        # RGBA_8888 always has 8 bits per color, 4 channels.
+        image_data.add(8)
+        image_data.add(4)
+        # sizeInBytes() is preferred, but PyQt complains it's a private method.
```

Comment:
Reported upstream as well:

https://www.riverbankcomputing.com/pipermail/pyqt/2020-May/042919.html

Reply:
Should now [be fixed](https://www.riverbankcomputing.com/pipermail/pyqt/2020-May/042920.html) with the next SIP version - so I suggest you do something like:

```python3
try:
    size = qimage.sizeInBytes()
except TypeError:
    # WORKAROUND for https://www.riverbankcomputing.com/pipermail/pyqt/2020-May/042919.html
    # byteCount() is obsolete, but sizeInBytes() is only available with SIP >= 5.3.0
    size = qimage.byteCount()
```